### PR TITLE
Minor changes to chapter 4

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR also changes very little, mainly changing the order of two code blocks to be clearer and fixing one technically incorrect example.

### CODE REVIEW

File: book/html.md [hunk 1]
```diff
@@ -245,11 +237,15 @@ works!
 ::: {.further}
 The ill-considered Javascript `document.write` method allows
 Javascript to modify the HTML source code while it's being parsed!
-Modern browsers use [speculative][speculative-parsing] parsing to
-make this fast and avoid evaluating Javascript while parsing.
+This is actually a [bad idea][document-write-bad]. A naive
+implementation of `document.write` would have the HTML parser stop to
+execute JavaScript, but that would be very slow. In fact, modern
```
Comment (chrishtr) [inline book/html.md:242]:
```suggestion
execute JavaScript, but that would slow down loading of images, JavaScript and so on referenced later in the page. In fact, modern
```

File: book/html.md [hunk 2]
```diff
@@ -245,11 +237,15 @@ works!
...
+browsers use [speculative][speculative-parsing] parsing to make this
+fast and avoid evaluating Javascript while parsing.
```
Comment (chrishtr) [inline book/html.md:244]: "
```suggestion
fast.
```

This phrase is not accurate / meaningless..

Comment (chrishtr) [reply book/html.md:244]: In general the parser does actually stop for all scripts. The speculative parser is just to find subresources to load faster.

File: book/html.md [hunk 3]
```diff
@@ -53,23 +50,19 @@ class Element:
         self.parent = parent
 ```
 
-I added a `children` field to both `Text` and `Element`, even though
```
Comment (chrishtr) [inline book/html.md:56]: Why did you remove this and the footnote above?

Comment (pavpanchekha) [reply book/html.md:56]: Uhh, I meant to remove exactly one.

File: book/html.md [hunk 4]
```diff
@@ -239,9 +242,10 @@ The ill-considered Javascript `document.write` method allows
 Javascript to modify the HTML source code while it's being parsed!
 This is actually a [bad idea][document-write-bad]. A naive
```
Comment (chrishtr) [inline book/html.md:243]:
```suggestion
This is actually a [bad idea][document-write-bad].
```

File: book/html.md [hunk 5]
```diff
@@ -239,9 +242,10 @@ The ill-considered Javascript `document.write` method allows
...
 implementation of `document.write` would have the HTML parser stop to
```
Comment (chrishtr) [inline book/html.md:244]:
```suggestion
An implementation of `document.write` must have the HTML parser stop to
```

**Review by chrishtr — APPROVED** [review]
